docs/builtins.rst

```diff
@@ -0,0 +1,304 @@
+
+      more-users = But ultimately, { DROP($users, 1) } like your message as well.
+
+
+Implicit use
+============
+
+In order to simplify most common scenarios, L20n will run some default
+functions while resolving placeables.
+
+For the list of implicit functions, the implict example has exactly the same
+result as the explicit one.
+
+``NUMBER``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `NUMBER` function on it.
+
+    Example::
+
+      emails = Number of unread emails { $unreadEmails }
+
+      emails2 = Number of unread emails { NUMBER($undeadEmails) }
+
+``DATETIME``
+    If the variable passed from the developer is a date and is used in
+    a placeable, L20n will implicitly call a `DATE` function on it.
+
+    Example::
+
+      log-time = Entry time: { $date }
+
+      log-time2 = Entry time: { DATETIME($date) }
+
+``PLURAL``
+    If the variable passed from the developer is a number and is used in
+    a selector expression, L20n will implicitly call a `PLURAL` function on it.
+
+    Example::
+
+      liked-count = { PLURAL($num) ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+      liked-count2 = { $num ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+``LIST``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `LIST` function on it.
+
+    Also, if the placeable is a list of variables, L20n will implicitly
+    call a `LIST` function on it.
+
+    Example::
+
+      users = { LIST($user1, $user2, $user3) }
+
+      users2 = { $user1, $user2, $user3 }
+
+      users = { LIST($users) }
+
+      users2 = { $users }
+
+
+Functions as arguments
+============================
+
+In most cases users will not have to call out Function explicitly, thanks
+to the implicit calls.
+
+The cases where implicit doesn't work will often come when the Function
+has to be called with additional parameters, but even then, majority
+of scenarios will require the parameters to be set by the developer and only
+in rare cases localizer will have to touch them.
+
+Developers can provide the variable already wrapped in Function as an
+argument.
+
+Example::
+
+  main.js:
+
+  let date = new Date();
+  let s = ctx.format('key1', {
+    day: Intl.MessageDateTimeArgument(date, {
+      weekday: 'long'
+    })
+  })
+
+  main.ftl:
+
+  key1 = Today is { $day }
+
+If the localizer decide that they have to modify the parameters, for example
+because the string doesn't fit in the UI, they can pass the variable
+to the same Function and overload parameters. Example::
+
+  main.ftl:
+
+  key1 = Today is { DATETIME($day, weekday: "short") }
+
+
+
+Gecko runtime specific functions
+================================
+
+At the moment Gecko runtime adds the following functions:
+
+``PLATFORM``
```

Comment:
The bug about the renaming hasn't yet been resolved, so it's up to you if you want to use the new name here or the old one.